openedx/core/process_warnings.py

```diff
@@ -0,0 +1,241 @@
+        )
+        for category, group_in_category, category_count in category_sorted_by_count:
+            html = Text(
+                u'<span class="count">{category}, count: {count}</span> '
+            ).format(category=HTML(category), count=HTML(category_count))
+            html_writer.start_section(html, klass=u"category")
+            locations_sorted_by_count = group_and_sort_by_sumof(
+                group_in_category, "high_location", "num"
+            )
+
+            for (
+                location,
+                group_in_location,
+                location_count,
+            ) in locations_sorted_by_count:
+                # pp.pprint(location)
+                html = Text(
+                    u'<span class="count">{location}, count: {count}</span> '
+                ).format(location=HTML(location), count=HTML(location_count))
+                html_writer.start_section(html, klass=u"location")
+                message_group_sorted_by_count = group_and_sort_by_sumof(
+                    group_in_location, "message", "num"
+                )
+                # pdb.set_trace()
+                for (
+                    message,
+                    message_group,
+                    message_count,
+                ) in message_group_sorted_by_count:
+                    # pp.pprint(warning_text)
+                    html = Text(
+                        u'<span class="count">{warning_text}, count: {count}</span> '
+                    ).format(warning_text=HTML(message), count=HTML(message_count))
+                    html_writer.start_section(html, klass=u"warning_text")
+                    # warnings_object[location][warning_text] is a list
+                    for _, warning in message_group.iterrows():
+                        # pp.pprint(warning)
+                        html = Text(
+                            u'<span class="count">{warning_file_path}</span> '
+                        ).format(warning_file_path=HTML(warning["filename"]))
+                        html_writer.start_section(html, klass=u"warning")
+
+                        html = Text(u'<p class="lineno">lineno: {lineno}</p> ').format(
+                            lineno=HTML(warning["lineno"])
+                        )
+                        html_writer.write(html)
+                        html = Text(u'<p class="num">num_occur: {num}</p> ').format(
+                            num=HTML(warning["num"])
+                        )
+                        html_writer.write(html)
+
+                        html_writer.end_section()
+                    html_writer.end_section()
+                html_writer.end_section()
+            html_writer.end_section()
+
+
+parser = argparse.ArgumentParser(
+    description="Process and categorize pytest warnings and output html report."
+)
+parser.add_argument("--dir_path", default="test_root/log")
+parser.add_argument("--html_path", default="test_html.html")
```

Comment:
I think the normal convention is to use dashes rather than underscores in arguments like this.